### DISCUSSION TITLE
Angular arbitrary values doesn't work locally

### DISCUSSION THREAD

**Discussion by RnhT01** [tailwindlabs/tailwindcss#9065] [Help]
I am using Tailwindcss version 3.x.x.I followed this tutorial : https://tailwindcss.com/docs/guides/angular and I installed it.
Here is my html code
`<div class="ml-[{{type.level * 10}}px]"></div>`
It doesn't work, but when I use :
`<script src="https://cdn.tailwindcss.com"></script>`
It works. What is the problem?

**Comment by wongjn** [comment]
This is because of the time at which class scanning happens.

For the CDN, it scans the HTML in the page, and at which point, the `ml-[{{type.level * 10}}px]` class has already been rendered by Angular into `ml-[10px]` (or equivalent).

On the other hand, when using Tailwind in your build system, it encounters `ml-[{{type.level * 10}}px]` in the source code **literally** and thus will not generate any classes – see the [documentation on dynamic class names](https://tailwindcss.com/docs/content-configuration#dynamic-class-names).

If you still want your dynamic behavior of `margin-left` to be equal to 10 times the `type.level`, we can use CSS variables like so (forgive me if my syntax is incorrect, I'm not very familiar with Angular!):
```html
<div class="ml-[var(--margin-left)]" style="--margin-left:{{type.level * 10}}px"></div>
```

Also in this case, it is a bit redundant and one may wish to use `margin-left` style directly and forego Tailwind:

```html
<div style="margin-left:{{type.level * 10}}px"></div>
```
